test/argsPass.js

```diff
@@ -0,0 +1,175 @@
+import { stub } from 'sinon';
+import { all, any, none } from 'ramda';
+import { assert } from 'chai';
+
+import { argsPass } from '../src/index';
+
+describe('argsPass', function() {
+  let p1False;
+  let p1True;
+  let p2True;
+  let p2False;
+  let p3True;
+  let p3False;
+  let c2True;
+  let c1True;
+  let c2False;
+  let c1False;
+
+  beforeEach(function() {
+    p1False = stub().returns(false);
+    p1True = stub().returns(true);
+    p2False = stub().returns(false);
+    p2True = stub().returns(true);
+    p3False = stub().returns(false);
+    p3True = stub().returns(true);
+    c2True = stub().returns(true);
+    c1True = stub().returns(c2True);
+    c2False = stub().returns(false);
+    c1False = stub().returns(c2False);
+  });
+
+  it('returns a new curried function for each argument supplied', function() {
+    const f = argsPass(c1True, [p1True, p2True, p3True]);
+    assert.isTrue(f(1)(2)(3));
+    assert.isTrue(p1True.calledWith(1));
+    assert.isTrue(p2True.calledWith(2));
+    assert.isTrue(p3True.calledWith(3));
+    assert.isTrue(c2True.calledWith([true, true, true]));
+    assert.isTrue(p1True.calledOnce);
+    assert.isTrue(p1True.calledOnce);
+    assert.isTrue(p2True.calledOnce);
+    assert.isTrue(c1True.calledOnce);
+    assert.isTrue(c2True.calledOnce);
+  });
+
+  context('with same number of functions as arguments', function() {
+    context('with all functions returning truthy values', function() {
+      specify('returns true', function() {
+        const f = argsPass(c1True, [p1True, p2True, p3True]);
+        assert.isTrue(f(1, 2, 3));
+        assert.isTrue(p1True.calledWith(1));
+        assert.isTrue(p2True.calledWith(2));
+        assert.isTrue(p3True.calledWith(3));
+        assert.isTrue(c2True.calledWith([true, true, true]));
+        assert.isTrue(p1True.calledOnce);
+        assert.isTrue(p2True.calledOnce);
+        assert.isTrue(p3True.calledOnce);
+        assert.isTrue(c1True.calledOnce);
+        assert.isTrue(c2True.calledOnce);
+      });
+    });
+
+    context('with one function returning a falsy value', function() {
```

Comment:
Missing test for currying

Reply:
Yes we should separate the scope of the tests. Test should explicitly tell us what it is testing and should test for one thing. Currying in this test is implicit. So please extract to separate `it/context/specify`.

Reply:
@char0n Sorry. I'm confused. There is already an (explicit)  test for currying on line 32:
```
it('should return a new curried function for each argument supplied', function() {
```
The test linked above is not using currying and is testing for something else. 

I'm happy to add more tests, but I'm unclear of what exactly you'd like me to test for given that currying is already covered.

Reply:
Right. Sorry read the code in hurry. PR is fine. Go for a merge.